Sloth.Core/Domain/Transaction.cs

```diff
@@ -25,7 +29,8 @@ public Transaction()
         [NotMapped]
         public string CreatorName => Creator?.UserName;
 
-        public string Status { get; set; }
+        // Status updates must go through SetStatus to ensure StatusEvents are properly updated
+        public string Status { get; private set; }
```

Comment:
@sprucely Dumb question... Is this going to get set when read from the db? I'm assuming yes,  but I can't remember.

Reply:
Yep. EF can write to private members.